src/TraceEvent/ETWTraceEventSource.cs

```diff
@@ -455,9 +455,9 @@ private void Initialize(string fileOrSessionName, TraceEventSourceType type)
                 sessionEndTimeQPC = this.UTCDateTimeToQPC(maxSessionEndTimeUTC);
             }
             Debug.Assert(_QPCFreq != 0);
-            if (pointerSize == 0)       // Real time does not set this (grrr). 
+            if (pointerSize == 0 || IsRealTime)  // We get on x64 OS 4 as pointer size which is wrong for realtime sessions. Fix it up. 
```

Comment:
Yes I have checked that on x64 I have got 4 as pointer size back. I guess that can happen if you start tracing from an x86 process but I have not looked deeper but the  set value was definitely wrong.